catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrMetacardClientImpl.java

```diff
@@ -296,11 +287,36 @@ protected SolrQuery postAdapt(QueryRequest request, SolrFilterDelegate filterDel
         query.setStart(request.getQuery()
                 .getStartIndex() - 1);
 
+        if (queryingForAllRecords(request)) {
+            try {
+                int numRows = queryForNumberOfRows(query);
+                query.setRows(numRows);
+            } catch (SolrServerException | IOException | SolrException | ArithmeticException exception) {
+                throw new UnsupportedQueryException("Could not complete solr query.", exception);
+            }
+        } else {
+            query.setRows(request.getQuery()
+                    .getPageSize());
+        }
+
         setSortProperty(request, query, filterDelegate);
 
         return query;
     }
 
+    private boolean queryingForAllRecords(QueryRequest request) {
+        return request.getQuery()
+                .getPageSize() < 1;
+    }
+
+    private int queryForNumberOfRows(SolrQuery query) throws SolrServerException, IOException {
+        query.setRows(0);
+        QueryResponse solrResponse = client.query(query, SolrRequest.METHOD.POST);
```

Comment:
Talked to @brjeter and we'll try to see if running the original query will impact the overall processing time too much.